src/cbc_sdk/platform/base.py

```diff
@@ -957,6 +959,10 @@ def _search(self, start=0, rows=0):
             self._processed_segments = result.get("processed_segments", 0)
             self._count_valid = True
             if self._processed_segments != self._total_segments:
+                retry_counter = 0 if self._processed_segments > last_processed_segments else retry_counter + 1
```

Comment:
There probably needs to be some level of delay/sleep otherwise the 10 iterations will go too quickly. This does need to take at a minimum 10-15 seconds

Reply:
Addressed.